src/cclib/io/ccio.py

```diff
@@ -355,29 +356,33 @@ def _determine_output_format(outputtype, outputdest):
     #  1. outputtype
     #  2. outputdest
 
+    outputclasses = {
+        'cjson': cjsonwriter.CJSON,
+        'json': cjsonwriter.CJSON,
+        'cml': cmlwriter.CML,
+        'xyz': xyzwriter.XYZ,
+        'molden': moldenwriter.MOLDEN
+    }
+
     # First check outputtype.
     if isinstance(outputtype, str):
-        if outputtype.lower() in ('cjson', 'json'):
-            outputclass = cjsonwriter.CJSON
-        elif outputtype.lower() == 'cml':
-            outputclass = cmlwriter.CML
-        elif outputtype.lower() == 'xyz':
-            outputclass = xyzwriter.XYZ
+        try:
+            outputclass = outputclasses[outputtype.lower()]
+        except:
+            # Determine the outputclass using outputdest in else part.
+            pass
     else:
         # Then checkout outputdest.
         if isinstance(outputdest, str):
-            extension = os.path.splitext(outputdest)[1]
+            extension = os.path.splitext(outputdest)[1].lower()
         elif isinstance(outputdest, fileclass):
-            extension = os.path.splitext(outputdest.name)[1]
+            extension = os.path.splitext(outputdest.name)[1].lower()
         else:
             raise ValueError
-        if extension.lower() in ('.cjson', '.json'):
-            outputclass = cjsonwriter.CJSON
-        elif extension.lower() == '.cml':
-            outputclass = cmlwriter.CML
-        elif extension.lower() == '.xyz':
-            outputclass = xyzwriter.XYZ
-        else:
+        try:
+            print (extension[1:])
+            outputclass = outputclasses[extension[1:]]
+        except:
             raise ValueError
```

Comment:
Now a `KeyError` is being raised with a custom messege `Unsupported Extension <ext>`.